### PR TITLE
Implement Runtime CR deletion handling

### PR DESCRIPTION
**Pull request by m00g3n**
**Description**

Implement `Runtime` CR deletion handling.

**Related issue(s)**

- https://github.com/kyma-project/infrastructure-manager/issues/209

### CODE REVIEW

File: internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go [hunk 1]
```diff
@@ -2,14 +2,47 @@ package fsm
...
 	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func sFnDeleteShoot(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
+	if !isGardenerCloudDelConfirmationSet(s.shoot.Annotations) {
```
Comment (koala7659) [inline internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go:15]: The whole section inside this if is a very good candidate for method exstraction. 

if !isGardenerCloudDelConfirmationSet 
   return PatchShootWithAnnotatiuon(s.shoot)

File: internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go [hunk 2]
```diff
@@ -2,14 +2,47 @@ package fsm
...
+				err.Error())
+			return updateStatusAndStop()
+		}
+		// see: https://gardener.cloud/docs/gardener/projects/#four-eyes-principle-for-resource-deletion
+		if s.shoot, err = m.ShootClient.Patch(ctx, s.shoot.Name, types.ApplyPatchType, shootData,
+			metav1.PatchOptions{
+				FieldManager: "kim",
+				Force:        ptr.To(true),
+			}); err != nil {
+			m.log.Error(err, "unable to patch shoot:")
```
Comment (koala7659) [inline internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go:39]: Please add shoot name to the logged  error message

File: internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go (hunk 1 above)
Comment (koala7659) [inline internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go:15]: Add following at the beginning of the function for better logging

	m.log.Info("Delete shoot state")

File: internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go [hunk 3]
```diff
@@ -32,5 +65,22 @@ func sFnDeleteShoot(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl
 		)
 		return updateStatusAndRequeue()
 	}
-	return updateStatusAndStop()
+
+	return updateStatusAndRequeue()
```
Comment (koala7659) [inline internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go:69]:
```suggestion
	return updateStatusAndRequeueAfter(gardenerRequeueDuration)
```

File: internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go [hunk 4]
```diff
@@ -32,5 +65,22 @@ func sFnDeleteShoot(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl
 		)
 		return updateStatusAndRequeue()
```
Comment (koala7659) [inline internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go:66]:
```suggestion
		return updateStatusAndRequeueAfter(gardenerRequeueDuration)
```

File: internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go (hunk 1 above)
Comment (m00g3n) [reply internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go:16]: As we agreed, we will be changing `Gardener` client in [this PR](https://github.com/kyma-project/infrastructure-manager/pull/275); refactor may be done there.

**Review by koala7659 — APPROVED** [review]
LGTM

**Review by koala7659 — APPROVED** [review]
LGTM

**Pull request merged**